libkbfs/folder_branch_ops.go

```diff
@@ -509,6 +509,45 @@ func (fbo *folderBranchOps) setBranchIDLocked(lState *lockState, bid BranchID) {
 	}
 }
 
+// getJournalPredecessorRevision returns the revision that precedes
+// the current journal head if journaling enabled and there are
+// unflushed MD updates; otherwise it returns
+// MetadataRevisionUninitialized.
+func (fbo *folderBranchOps) getJournalPredecessorRevision(ctx context.Context) (
+	MetadataRevision, error) {
+	jServer, err := GetJournalServer(fbo.config)
+	if err != nil {
+		// Journaling is disabled entirely.
+		return MetadataRevisionUninitialized, nil
+	}
+
+	jStatus, err := jServer.JournalStatus(fbo.id())
+	if err != nil {
+		// Journaling is disabled for this TLF, so use the local head.
+		// TODO: JournalStatus could return other errors (likely
+		// file/disk corruption) that indicate a real problem, so it
+		// might be nice to type those errors so we can distinguish
+		// them.
+		return MetadataRevisionUninitialized, nil
+	}
+
+	if jStatus.BranchID != NullBranchID.String() {
+		return MetadataRevisionUninitialized,
+			errors.New("Cannot find most recent merged revision while staged")
+	}
+
+	if jStatus.RevisionStart == MetadataRevisionUninitialized {
+		// The journal is empty, so the local head must be the most recent.
+		return MetadataRevisionUninitialized, nil
+	} else if jStatus.RevisionStart == MetadataRevisionInitial {
+		// Nothing has been flushed to the servers yet, so don't
+		// return anything.
+		return MetadataRevisionUninitialized, errors.New("No flushed MDs yet")
```

Comment:
I'm not sure returning an error here is correct -- it seems like if a user creates a new TLF and writes a single rev to the journal, then restarts, then when it tries to set the initial head, it will encounter an error. Am I missing something?